API_MANIFESTO.md

```diff
@@ -0,0 +1,91 @@
+# The API Manifesto
+
+# Infinispan API dependency module
+
+The ```infinispan-api``` dependency will contain the public API. This module will be independent from the other modules. 
+Dependency with implementations will be **logical**. This will keep the modularity needed for the different use cases.
+
+Examples:
+ * Cache use case
+ 
+ * NoSQL data store use case
+ 
+ * Tasks, concurrency, counters uses cases
+
+The dependencies to thirty-party libraries must be avoided. This module should contain dependencies to JDK classes, 
```

Comment:
Are sure we want to avoid `javax.transaction`, too?

Reply:
If it's up to me, yes

Reply:
if it's up to me, YES

Reply:
This might not always be possible, e.g. reactive streams. Of course, now that we're in Java 11 we don't need the reactive streams dependency.

Maybe if an API requires a 3rd party dependency, it should be defined as an independent module and have some way of getting that module via the Infinispan API? (I know, I can't let my idea go! 😉 )

Reply:
ok

Reply:
@galderz Concerning reactive streams I added it because of java 8 but since we consider this API should be for Java 11 only, of course, it's not needed and I will remove it. 

For the future, I'm going to add the following to the document:

If you really *need* for a reason to use an external library to be part of an API. Here is a non-exhaustive list of what we should discuss *before promoting* an API that uses an external dependency:
1) Are we are being too opinionated? 
2) Is there a way to keep that library in implementation instead of public API? Creating our Infinispan public classes and keeping it internal
3) Who are we depending on? What is going to happen when this library won't be maintained or will break? 
4) Is this library such important that is considered to be included in the JDK in the near future? 

Promotion to public official API should be done by care and debate. By default we don't use other classes than JDK + ours.

Reply:
Can you clarify? This is going to be JDK11+ only?

Reply:
@karesti sorry I missed that the document was general and not specific to the reactive API.